Deprecate StringConcatenateAssignmentNode (#183)
eisop/checker-framework: docs/CHANGELOG.md

visibility. It is only meant as a utility method for use within a
transfer function.

> Deprecated class `StringConcatenateAssignmentNode` and its related usage.

Reviewer: ```suggestion
Deprecated class `StringConcatenateAssignmentNode` and its usages.
```

Author: Thanks, done.

> `CFGTranslationPhaseOne` now generates an assignment and concatenation instead.
> 

Reviewer: ```suggestion
`CFGTranslationPhaseOne` now generates an assignment and a concatenation node instead.
This avoids error prone duplication of logic.
```

Author: Thanks, done.

visibility. It is only meant as a utility method for use within a
transfer function.

Deprecated class `StringConcatenateAssignmentNode` and its usages.

> `CFGTranslationPhaseOne` now generates an assignment and a concatenation node instead.

Reviewer: ```suggestion
String concatenate assignments are now desugared to an assignment and a concatenation node instead.
```
